csharp/src/Ice/LocatorInfo.cs

```diff
@@ -229,6 +238,75 @@ internal void ClearCache(Reference reference)
             return (endpoints, cached);
         }
 
+        // Encoding represents the encoding used to send the getRegistry request.
+        internal async ValueTask<ILocatorRegistryPrx?> GetLocatorRegistryAsync(Encoding encoding)
+        {
+            if (_locatorRegistry == null)
+            {
+                ILocatorRegistryPrx? prx =
+                    await Locator.Clone(encoding: encoding).GetRegistryAsync().ConfigureAwait(false);
+
+                // The locator registry can't be located and we use ordered endpoint selection in case the locator
+                // returned a proxy with some endpoints which are preferred to be tried first.
```

Comment:
I would simplify this and allow the proxy to be located. I wonder also if the endpoint selection type should be an encoded proxy option or if the default should be ordered for proxies received over the wire (and possibly random for stringified proxies).

Reply:
For a future PR. I didn't change this code at all.